packages/runtime/container-runtime/src/opLifecycle/remoteMessageProcessor.ts

```diff
@@ -0,0 +1,94 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ISequencedDocumentMessage, MessageType } from "@fluidframework/protocol-definitions";
+import { ContainerMessageType, ContainerRuntimeMessage } from "../containerRuntime";
+import { OpDecompressor } from "./opDecompressor";
+import { OpSplitter } from "./opSplitter";
+
+export class RemoteMessageProcessor {
+    constructor(
+        private readonly opSplitter: OpSplitter,
+        private readonly opDecompressor: OpDecompressor,
+    ) { }
+
+    public get hasPartialMessages(): boolean {
+        return this.opSplitter.hasChunks;
+    }
+
+    public get partialMessages(): ReadonlyMap<string, string[]> {
```

Comment:
I'd rather push responsibility to generate snapshots down. I.e. ContainerRuntime calls snapshot() method on this instance, and it reruns either blob/tree, or undefined. This encapsulated it better, requires less APIs, and ensure format (parsing and serialization are next to each other and controlled by whoever holds data